config/routes.rb

```diff
@@ -66,11 +66,32 @@
         end
       end
 
-      resources :allocations, only: %i[new], controller: 'allocations'
+      constraints lambda {
+        |request| PrisonService.womens_prison?(request.path_parameters.fetch(:prison_id))
+      } do
```

Comment:
Might be nice to have a comment to make this extra clear:

```suggestion
      constraints lambda {
        # Women's allocation routes
        |request| PrisonService.womens_prison?(request.path_parameters.fetch(:prison_id))
      } do
```